sources/metricdog/src/host_check.rs

```diff
@@ -0,0 +1,15 @@
+use crate::error::Result;
+
+pub(crate) trait HostCheck {
+    /// Checks whether this current boot is the first boot of the host.
+    fn is_first_boot(&self) -> Result<bool>;
+
+    /// The time, in milliseconds, it took for the host to reach the 'preconfigured' stage.
+    fn preconfigured_time(&self) -> Result<String>;
```

Comment:
nit: in general, time fields without an associated unit are harder to reason about:
```suggestion
    /// The time, in milliseconds, it took for the host to reach the 'preconfigured' stage.
    fn preconfigured_time_millis(&self) -> Result<String>;
```